### PR TITLE
get_challengeId_from_props

### LINKED ISSUE #87: IndividualChallenge - get challengeId from props - (replace hard-coded challenge)

### PR DESCRIPTION
**Pull request by chem2016**
Updating individual challenge client component, to get the challenges from store instead of hard coding it. 

close #87

### CODE REVIEW

File: client/components/IndividualChallenge.js [hunk 1]
```diff
@@ -84,9 +96,12 @@ const _IndividualChallenge = props => {
 const mapDispatchToProps = dispatch => ({
   putUserchallenge: (userAnswer, userchallengeId, isSubmit) =>
```
Comment (theo333) [inline client/components/IndividualChallenge.js]:
```suggestion
  updateUserchallenge: (userAnswer, userchallengeId, isSubmit) =>
```

File: client/components/IndividualChallenge.js [hunk 2]
```diff
@@ -84,9 +96,12 @@ const _IndividualChallenge = props => {
...
     dispatch(putUserchallenge(userAnswer, userchallengeId, isSubmit)),
```
Comment (theo333) [inline client/components/IndividualChallenge.js]: Obviously need to change the name to `updateUserchallenge`  in the store and also elsewhere in this file.
```suggestion
    dispatch(updateUserchallenge(userAnswer, userchallengeId, isSubmit)),
```

File: client/components/IndividualChallenge.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
 import React, { useState, useEffect } from 'react';
 import CodeMirror from 'react-codemirror';
 import { connect } from 'react-redux';
-import { putUserchallenge } from '../store/userchallenge';
+import { putUserchallenge, fetchChallenges } from '../store';
 
 const _IndividualChallenge = props => {
+  useEffect(() => {
+    props.fetchChallenges();
```
[the remark this answers is not in the document]
Comment (chem2016) [reply client/components/IndividualChallenge.js]: Yes, I am using the thunk method fetch challenges which fetch all the challenges. What I can do is to filter it in the mapStateToProps or creating another thunk method called fetchOneChallenge.

**Comment by chem2016** [comment]
@theo333 , comments have been addressed. Please see diff.

File: client/components/IndividualChallenge.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (theo333) [reply client/components/IndividualChallenge.js]: I think creating a `fetchOneChallenge` thunk makes the most sense.  No point in grabbing all of them each time the component mounts.

File: client/components/IndividualChallenge.js [hunk 4]
```diff
@@ -16,24 +20,32 @@ const _IndividualChallenge = props => {
   const putValue = isSubmit => {
```
Comment (theo333) [inline client/components/IndividualChallenge.js]: Missed this the first time.  Also update this in Run and Submit buttons.
```suggestion
  const updateValue = isSubmit => {
```

**Comment by chem2016** [comment]
@theo333 , comments addressed.

File: client/components/IndividualChallenge.js [hunk 5]
```diff
@@ -13,27 +17,36 @@ const _IndividualChallenge = props => {
...
       .catch(ex => console.log(ex));
   };
 
-  const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
   const options = {
     lineNumbers: true,
     mode: 'javascript',
   };
+
+  if (!Object.keys(props.challenge).length) {
```
Comment (hgranthorner) [inline client/components/IndividualChallenge.js]: I think this should be written out, it's a little hard to read as is.

`if (Object.keys(props.challenge).length === 0) {`

**Comment by chem2016** [comment]
@hgranthorner , comments address. Please check!

**Pull request merged**